File: pubnub-java/src/main/java/com/pubnub/api/endpoints/push/ListPushProvisions.java
package com.pubnub.api.endpoints.push;

import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.push.PNPushListProvisionsResult;

public interface ListPushProvisions extends Endpoint<PNPushListProvisionsResult> {
    ListPushProvisions pushType(com.pubnub.api.enums.PNPushType pushType);

    ListPushProvisions deviceId(String deviceId);

    ListPushProvisions environment(com.pubnub.api.enums.PNPushEnvironment environment);

    ListPushProvisions topic(String topic);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/push/ListPushProvisions.kt
package com.pubnub.api.endpoints.push

import com.pubnub.api.Endpoint
import com.pubnub.api.enums.PNPushEnvironment
import com.pubnub.api.enums.PNPushType
import com.pubnub.api.models.consumer.push.PNPushListProvisionsResult

/**
 * @see [PubNub.auditPushChannelProvisions]
 */
actual interface ListPushProvisions : Endpoint<PNPushListProvisionsResult> {
    val pushType: PNPushType
    val deviceId: String
    val topic: String?
    val environment: PNPushEnvironment
}
